content/blog/2019/secure-egress-control-istio/index.md

```diff
@@ -0,0 +1,323 @@
+The attackers, on the contrary, want to access external services unmonitored, so the attack will not be detected.
+The second goal of the attackers is to access `mongo1.composedb.com`, which is forbidden for the application A. Istio
+must enforce the policy that forbids access of application A to `mongo1.composedb.com` and must prevent the attack.
+
+Now let's see which attacks the attackers will try to perform to achieve their goals and how Istio secure egress traffic
+control will prevent each kind of attack. The attackers may try to:
+
+1. **Bypass** the container's sidecar proxy and access external services directly. This attack is prevented by a
+   Kubernetes Network Policy or by an L3 firewall that allow egress traffic to exit the mesh only from the egress
+   gateway.
+1. **Compromise** the egress gateway and force it to send fake information to the monitoring system or to disable
+   enforcement of the security policies.
+   This attack is prevented by applying the special security measures to the egress gateway pod.
+1. Since the previous attacks are prevented, the attackers have no other option but to direct the traffic through the
+   egress gateway. The traffic will be monitored by the egress gateway, so the goal of the attackers to access
+   external services unmonitored cannot be achieved. The attackers may want to try to achieve their second goal, that is
+   to access `mongo1.composedb.com`. To achieve it, they may try to **impersonate** as the application B since the
+   application B is allowed to access `mongo1.composedb.com`. This attack, fortunately, is prevented by Istio's [strong
+   identity support](/docs/concepts/security/#istio-identity).
+
+After I showed how Istio egress traffic control can prevent possible attacks, let me describe its advantages over
+Kubernetes Network Policies and legacy egress proxies/firewalls.
+
+### Advantages of Istio egress traffic control
+
+Istio egress traffic control is **DNS-aware**: you can define policies based on URLs or on wildcard domains like
+`*.ibm.com`. In this sense, it is better than Kubernetes Network Policies which are not DNS-aware.
+
+Istio egress traffic control is **transparent** with regard to TLS traffic, since Istio is transparent:
+you do not need to change the applications or to configure their containers.
+For HTTP traffic with TLS origination, DevOps people must configure the applications to use HTTP instead of HTTPS
+when the applications run with Istio sidecars injected.
+
+Istio egress traffic control is **Kubernetes-aware**: the identity of the source of egress traffic is based on
+Kubernetes service accounts. Istio egress traffic control is better than the legacy DNS-aware proxies/firewalls which
+are not transparent and not Kubernetes-aware.
+
+Istio egress traffic control is **secure**: it is based on the strong identity of Istio and, when the cluster operators
```

Comment:
based on Istio's strong security model

Reply:
@geeknoid _strong identity_ is not in the Istio's glossary. Do you think we should add it?

See https://preliminary.istio.io/docs/concepts/security/
> The Istio security features provide strong identity

I checked that it is also used outside of Istio, for example:

https://www.ubisecure.com/access-management/privileged-access-management/
> Strong Identity with Privileged Access Management

I use a link to  https://preliminary.istio.io/docs/concepts/security/#istio-identity I mention _strong identity_ for the first time earlier.

Reply:
Yeah, it's worth adding to the glossary. You can then use `{{<gloss>}}strong identity{{</gloss>}}` to let the user bring up a popup with the definition.

Reply:
Added an issue to add strong identity to glossary https://github.com/istio/istio.io/issues/3434